### ISSUE TITLE
Looking at the code, I see that there is an dequantisation process when actually doing the inference, i.e. the actual matrix multiplication is done with floating point arithmetic right?

### ISSUE THREAD

**Issue by dingjingzhen** [mit-han-lab/llm-awq#43]
Looking at the code, I see that there is an dequantisation process when actually doing the inference, i.e. the actual matrix multiplication is done with floating point arithmetic right?

**Comment by casperbh96** [comment]
Yes, every time you load the model, it replaces some layers with their custom layers that uses the CUDA kernels. Only the quantized layers run in INT4 while the rest run in FP16.
